### ISSUE TITLE
File provisioner does not transfer folders

### ISSUE THREAD

**Issue by rnalrd** [hashicorp/packer-plugin-lxc#31]
In this example,

`
provisioner "file" {
  sources = [ "srcfolder/foo" ] destination = "dstfolder"
}
`

files are copied in "dstfolder" instead of "dstfolder/foo".

**Comment by devashish-patel** [comment]
Hi, @rnalrd thanks for reaching out!

I tried to reproduce the issue with 2 scenarios...

1. The `destination` is already present
- I put `tmp` as the `destination` and the output was as expected. The files under `"srcfolder/foo"` directory were copied to `"tmp/foo"`. (The dir `foo` gets created by packer automatically).

- Example `build`:
```hcl
build {
...

  provisioner "file" {
    sources     = ["srcfolder/foo"]
    destination = "tmp"
  }

...
}
```

- Example output:
```bash
➜  packer_tutorial docker run -it 942522d4840e
# ls -la tmp
total 12
drwxrwxrwt 1 root root    4096 Jun 27 17:39 .
drwxr-xr-x 1 root root    4096 Jun 27 17:40 ..
drwxr-xr-x 2  503 dialout 4096 Jun 27 17:30 foo
# ls -la tmp/foo
total 12
drwxr-xr-x 2  503 dialout 4096 Jun 27 17:30 .
drwxrwxrwt 1 root root    4096 Jun 27 17:39 ..
-rw-r--r-- 1  503 dialout   13 Jun 27 17:30 test.txt
```

2. The `destination` is _not_ present
- I put `data` as the `destination` (note that the dir with the name `data` is not present by default) and I faced the same problem as you have mentioned in the issue.
- Although the reason behind that is when the `destination` dir is not present the packer expects us to create the dir using the `shell` provisioner before we copy the files to that location.
- I added that part to the `build` block and now the files are being copied to the right location.

- Example `build`:
```hcl
build {
...

  provisioner "shell" {
    inline = ["mkdir dstfolder"]
  }

  provisioner "file" {
    sources     = ["srcfolder/foo"]
    destination = "dstfolder"
  }

...
}
```

- Example output:
```
# ls -la data
total 12
drwxr-xr-x 3 root root    4096 Jun 27 17:59 .
drwxr-xr-x 1 root root    4096 Jun 27 20:28 ..
drwxr-xr-x 2  503 dialout 4096 Jun 27 17:30 foo
# ls -la data/foo
total 12
drwxr-xr-x 2  503 dialout 4096 Jun 27 17:30 .
drwxr-xr-x 3 root root    4096 Jun 27 17:59 ..
-rw-r--r-- 1  503 dialout   13 Jun 27 17:30 test.txt
```

Hope this resolves the problem otherwise feel free to continue the thread!

**Comment by rnalrd** [comment]
Thank you. Indeed re-reading the documentation says to create the destination directory first. I should have read better. :smile: 
Closing.

**Comment by devashish-patel** [comment]
Hello @rnalrd! 

 I think we have found a potential fix for this issue. Specifically for the `lxc` plugin, we had `/.` at the end of the `source` dir path in the Upload Dir function. That was causing the unexpected behavior. The fix resides in PR #39 and [here](https://app.circleci.com/pipelines/github/hashicorp/packer-plugin-lxc/74/workflows/180b31cf-17cd-474a-9ae5-cfc8302a5cd0/jobs/1009/artifacts) are some binaries to install manually.  

#### Steps to install the binaries manually:
- The easiest way to manually install a plugin is to name it correctly, then place it in the proper directory. To name a plugin correctly, make sure the binary is named `packer-plugin-NAME`. For example, `packer-plugin-lxc` for a "plugin"
binary named "lxc". This binary will make one or more components available to use. 

- Once the plugin is named properly, Packer automatically discovers plugins in the following directories in the given order. If a conflicting plugin is found later, it will take precedence over the one found earlier.

1.  The directory where `packer` is, or the executable directory.

2.  The `$HOME/.packer.d/plugins` directory, if `$HOME` is defined (unix)

3.  The `%APPDATA%/packer.d/plugins` if `%APPDATA%` is defined (windows)

4.  The `%USERPROFILE%/packer.d/plugins` if `%USERPROFILE%` is defined
    (windows)

5.  The current working directory.

6.  The directory defined in the env var `PACKER_PLUGIN_PATH`. There can be more
    than one directory defined; for example, `~/custom-dir-1:~/custom-dir-2`.
    Separate directories in the PATH string using a colon (`:`) on posix systems and
    a semicolon (`;`) on windows systems. The above example path would be able to
    find a provisioner named `packer-provisioner-foo` in either
    `~/custom-dir-1/packer-provisioner-foo` or
    `~/custom-dir-2/packer-provisioner-foo`.

- In simple words...
    - Download the suitable zip from the [artifactory](https://app.circleci.com/pipelines/github/hashicorp/packer-plugin-lxc/74/workflows/180b31cf-17cd-474a-9ae5-cfc8302a5cd0/jobs/1009/artifacts) 
    - Unzip it
    - Rename it to `packer-plugin-lxc`
    - Move it to the appropriate location as mentioned above and make sure to remove any existing instance of the `lxc` plugin installed earlier
    - Remove the `packer` block from your template
    - Run the build  


Could you please tell me if the fix works for you? :D

Please feel free to reach out if you need help!

Thanks!

**Comment by rnalrd** [comment]
Thank you very much for the fix. Unfortunately I'm unable to use the artifact because it's not statically linked and I'm running Alpine Linux which uses musl libc instead of GLIBC.

Could you provide a statically linked artifact?

**Comment by devashish-patel** [comment]
Hi, @rnalrd We have released the fix for this issue in **v1.0.2**, please update the version.

**Comment by rnalrd** [comment]
Thank you, I confirm this is fixed
